### DISCUSSION TITLE
Time to reconsider parent package support?

### DISCUSSION THREAD

**Discussion by ryenus** [npm/rfcs#270] [General]
I guess the initial request https://github.com/npm/npm/issues/8112 for parent package.json support
was not considered at a time npm was meant to receive minimal changes.

Given that npm has received quite many changes, and the intended use cases are still reasonable, such as

* specify a common set of dependencies, and/or devDependencies
* centrally manage the versions of the dependencies, like maven dependencyManagement
* reuse other configs, especially npm tasks and scripts, etc.

These are even more useful within the microservices context, where one might have similar setup for dozens of, if not hundreds of similar structured projects. 
And there's https://github.com/Cosium/dry-dry that looks quite interesting.

So maybe it's time to revisit the request and do something about it?

**Comment by ljharb** [comment]
See https://github.com/npm/rfcs/pull/165

**Reply by darcyclarke** [reply]
@ryenus as @ljharb has mentioned here, we are currently looking into the details of this over on our https://github.com/npm/rfcs repo. Feel free to add to that proposal if you have ideas (rendered ref. https://github.com/npm/rfcs/pull/165/files?short_path=28d904d#diff-28d904d78a1ed70fc704cdade059b6d187bfcf84ccef1a4304d3afd77681f336).

**Comment by mithatyagmuroglu** [comment]
heyyyyy

**Reply by bebo13133** [reply]
hi

**Comment by KIRAN-KUMAR-K3** [comment]
It appears that you are referring to GitHub issue #165. To provide more accurate and helpful information, I would need the context or details from the issue description or discussion. GitHub issues often contain specific details, comments, and conversations related to a particular topic. If you have a specific question or if there's something specific you would like assistance with regarding issue #165  please provide more details, and I'll do my best to assist you.

Dear @ryenus, 

As indicated by @ljharb in this discussion, our team is actively examining the specifics of the matter on our GitHub repository at https://github.com/npm/rfcs. Your contributions and ideas to the ongoing proposal process are highly encouraged. Please feel free to augment the discussion by providing your insights through the proposal, which can be found at https://github.com/npm/rfcs/pull/165/files?short_path=28d904d#diff-28d904d78a1ed70fc704cdade059b6d187bfcf84ccef1a4304d3afd77681f336.

Thank you for your engagement and collaboration.
